app/models/order.rb

```diff
@@ -32,4 +33,30 @@ def total_items
   def set_total
     update(total: line_items.to_a.sum(&:price))
   end
+
+  def received
+    update(status: 'received', placed_on: Time.now)
+  end
+
+  def check_branch_inventory?
+    meal_max_quantity = meals.joins(ingredients_meals: :inventories)
+          .where(inventories: { branch_id: branch_id })
+          .group(:id).minimum('inventories.quantity/ingredients_meals.ingredient_quantity')
+    line_items.joins(:meal).each do |line_item|
```

Comment:
move this logic to line_item model and create a method available and run on line items here. Any line item returning false should add error